src/proto/h2/client.rs

```diff
@@ -44,6 +44,7 @@ pub(crate) struct Config {
     pub(crate) keep_alive_timeout: Duration,
     #[cfg(feature = "runtime")]
     pub(crate) keep_alive_while_idle: bool,
+    pub(crate) max_concurrent_reset_streams: Option<usize>,
```

Comment:
Can this be zero in the first place?

Reply:
If this is zero, then it will override the default value in `h2`, which 10. We could copy the default value from `h2` here, what do you think?

Reply:
Err, that seemed very implicit which is probably a bad idea. Can it be `NonZeroUsize` or so if it's `None` we can default to 10 but when it's `Some(n)` it will be the same? I am asking about `Some(0)` for the case but not sure if we really want that.

Reply:
I am not entirely sure what you are proposing, maybe we can discuss it again? Just to understand: there are 3 possible config options that we try to make available to the user:

1. Not specified => the default configuration value from `h2` will be used,
2. Specified `> 0` => `h2` will use reset streams,
3. Specified `0` => `h2` will not use reset streams at all.

To make it less implicit, maybe we should export the default configuration from `h2` and use it here?

Reply:
That seemed a bit confusing but when I look at it again. I think it may be better if we just make it `usize`.

1. Not specified => default to `proto::DEFAULT_RESET_STREAM_MAX`
2. Specified => `h2` used specified (`0` will not use reset streams at all)

So we already have the default value set by default, I don't see why we need `Option` since it will always have a value.

Reply:
Perfect. This definitely works.

Reply:
I actually _would_ suggest that this be an `Option<usize>`, and if `None`, then the option just isn't set on the `h2` builder. That'd mean less of `h2` needs to be modified/exposed. We can just document "_a_ reasonable default is used", without saying what it is.

Reply:
@pickfire Considering the last comment of @seanmonstar, I would restore my previous commit if it is fine with you.

Reply:
@boxdot sure, @seanmonstar is the maintainer and I didn't do much here, reverting it is the best choice.